pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -445,6 +445,18 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 			// pod created, no more work possible for this round
 			continue
 		}
+
+		// If the Pod is in pending state then trigger PVC creation to create missing PVCs
+		if isPending(replicas[i]) {
+			klog.V(4).Infof(
+				"StatefulSet %s/%s is triggering PVC creation for pending Pod %s",
+				set.Namespace,
+				set.Name,
+				replicas[i].Name)
+			if err := ssc.podControl.CreatePersistentVolumeClaims(set, replicas[i]); err != nil {
```

Comment:
I don't think we can directly call `CreatePersistentVolumeClaims`. `CreateStatefulPod` also does additional PVC updates for the `StatefulSetAutoDeleletePVC` feature.

cc @mattcary

Reply:
any suggestions on how to include those updates here?

Reply:
Add a function to StatefulPodControl that creates the PVC if it does not exist.

Reply:
+1, UpdateStatefulPod below assumes the PVCs have been created, so I don't think it will correctly update if the PVC doesn't exist. 

Whatever you add to StatefulPodControl should probably involve a refactor of CreateStatefulPod as the logic will overlap.

Reply:
> Add a function to StatefulPodControl that creates the PVC if it does not exist.

I'm not sure I follow, doesn't `createPersistentVolumeClaims` already do this? It loops through a pod's pvcs defined in its spec and checks to see if they actually exist. If it doesn't exist, then it creates the pvc.

I think if i want to include the pvc updates for the auto delete pvc feature, all I have to do is copy the following from `CreateStatefulPod` to under this `createPersistentVolumeClaims` call I added:

```
if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
		// Set PVC policy as much as is possible at this point.
		if err := spc.UpdatePodClaimForRetentionPolicy(set, pod); err != nil {
			spc.recordPodEvent("update", set, pod, err)
			return err
		}
	}
```

Reply:
I think the suggestion is to create a variant of `CreateStatefulPod` that only handles creating the pvc and updating the retention policy.

Reply:
I created a new function in [this commit](https://github.com/kubernetes/kubernetes/pull/113270/commits/8c9867fbcffb27c0577b171aec91556ac24838a5)